Assets/PoisonMushroom.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PoisonMushroom : MonoBehaviour
+{
+    //Mushroom stats
+    public int health; //Mushroom health
+    public int damageNum; //amount of damage applied to Mushroom health
+    public float lifeTimer; //timer for destroying the Mushroom
+    private float generateTimer; //timer for generating the clouds
+    private BoxCollider collider;
+    private MeshRenderer mesh;
+
+    //Generating clouds using Particle System
+    private ParticleSystem cloud;
+    public float timeGenerate; //duration for generating the clouds
+
+    void Start()
+    {
+        collider = GetComponent<BoxCollider>();
+        mesh = GetComponent<MeshRenderer>();
+        cloud = GetComponent<ParticleSystem>();
+        generateTimer = timeGenerate;
+    }
+
+    void Update()
+    {
+        if (generateTimer > 0)
+        {
+            generateTimer -= Time.deltaTime;
+        }
+        else if (generateTimer == 0)
+        {
+            cloud.Stop();
+            cloud.Clear();
+        }
+        else if (generateTimer <= 0)
+        {
+            GeneratePoisonCloud();
+            generateTimer = timeGenerate;
+        }
+
+        //Start death sequence
+        if (health <= 0)
+        {
+            lifeTimer -= Time.deltaTime;
+            DestroyMushroom();
```

Comment:
this method should not be called here since its being called every frame in which the health of the mushroom is <=0 which is bad